### PR TITLE
helm: use correct namespace for provisioner in cephobjectstore template

### LINKED ISSUE #10153: rook-ceph-cluster chart wrong provisioner for obejctstore storageclass
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Provisioner for StorageClass of ObjectStore uses Release.Namespace instead of oeratorNamespace
https://github.com/rook/rook/blob/29c598bc36309ce8c3447aa5a7b2afc903eb2fe7/deploy/charts/rook-ceph-cluster/templates/cephobjectstore.yaml#L16

**Expected behavior:**
Use of operatorNamespace like StorageClass of Filesystem
https://github.com/rook/rook/blob/29c598bc36309ce8c3447aa5a7b2afc903eb2fe7/deploy/charts/rook-ceph-cluster/templates/cephfilesystem.yaml#L18

**How to reproduce it (minimal and precise):**

```
helm upgrade --install --namespace rook-ceph --set operatorNamespace=rook-ceph-system rook-ceph-cluster rook-stable/rook-ceph-cluster
```

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): `Flatcar Container Linux by Kinvolk 3139.2.0 (Oklo)`
* Kernel (e.g. `uname -a`): `Linux 5.15.32-flatcar SMP`
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): `rook: v1.8.9`
* Storage backend version (e.g. for ceph do `ceph -v`): `ceph version 16.2.7 (dd0603118f56ab514f133c8d2e3adfc983942503) pacific (stable)`
* Kubernetes version (use `kubectl version`): `v1.21.10`
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): `Rancher (rke)`
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): `HEALTH_OK`

### PR DESCRIPTION
**Pull request by bitfisher**
Signed-off-by: Markus Schmidleitner <markus.schmidleitner@ocilion.com>

**Description of your changes:**
Use correct namespace for provisioner in cephobjectstore template

**Which issue is resolved by this Pull Request:**
Resolves #10153

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — CHANGES_REQUESTED** [review]
Thanks for the fix. As shown in the CI failure now for the helm upgrade test, the challenge is that upgraded clusters will fail with the following error if the operator is in a different namespace from the cluster:
```
Error: UPGRADE FAILED: cannot patch "ceph-bucket-test-sc" with kind StorageClass: 
StorageClass.storage.k8s.io "ceph-bucket-test-sc" is invalid: provisioner: Forbidden: updates to provisioner are forbidden. , err -> exit status 1
```

To workaround this issue, I'd suggest the following:
1. Add a note to the [upgrade guide](https://github.com/rook/rook/blob/release-1.9/Documentation/ceph-upgrade.md#breaking-changes-in-this-release) that the object store class must be deleted before the upgrade if their operator is in a different namespace from the cluster and their object storage class is enabled.
2. The helm upgrade test must delete the object storage class before upgrade. To help navigate the tests, this might be best done in [ceph_upgrade_test.go](https://github.com/rook/rook/blob/master/tests/integration/ceph_upgrade_test.go#L124) on line 124 by copying line 121 from the test cleanup that deletes the object storage class:

```go
s.helper.BucketClient.DeleteBucketStorageClass(s.namespace, installer.ObjectStoreName, installer.ObjectStoreSCName, "Delete")
```

**Comment by bitfisher** [comment]
Unfortunately even after setting the correct namespace for provisioner reconciling OBC fails with `unsupported provisioner`.

See https://github.com/rook/rook/issues/5675#issuecomment-1109644166

**Comment by travisn** [comment]
@bitfisher From the discussion in #10153 it seems you got it working without the namespace prefix, right? So is this PR still needed?

**Comment by bitfisher** [comment]
Yes, without the namespace prefix it's working as expected, because the operator namespace is used automatically [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/bucket/util.go#L46).
But i think the chart still need to be adjusted because the Release-Namespace is used [there](https://github.com/rook/rook/blob/29c598bc36309ce8c3447aa5a7b2afc903eb2fe7/deploy/charts/rook-ceph-cluster/templates/cephobjectstore.yaml#L16).

**Review by travisn — COMMENTED** [review]
Thanks for the explanation, agreed this change is needed. Before merging we just need to have the merge conflict resolved, thanks.

File: Documentation/ceph-upgrade.md [hunk 1]
```diff
@@ -62,6 +62,9 @@ those releases.
   following the advice given in Rook's
   [v1.8 NFS docs](https://rook.github.io/docs/rook/v1.8/ceph-nfs-crd.html).
 
+* If you use Helm and your operator is in a different namespace from the cluster and object storage class is enabled
```
[the remark this answers is not in the document]
Comment (bitfisher) [reply Documentation/ceph-upgrade.md]: Rebased

**Review by travisn — APPROVED** [review]

**Pull request merged**